api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,65 @@
 package aws
 
-func (s *ServiceProvider) Delete(ServiceID string) error {
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+// Delete a specified service within a cluster.
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := int64(0)
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return err
+	}
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+	fqClusterID := addLayer0Prefix(s.Config.Instance(), environmentID)
+
```

Comment:
somewhere in here, we need to replicate the behavior of gathering the taskArns for a given service ala https://github.com/quintilesims/layer0/blob/develop/api/backend/ecs/service_manager.go#L115

(if the intention is to simplify that behavior somehow, lmk)

Reply:
Okay I wasn't sure about that, can we discuss it. I'm not sure it's necessary looking at the aws sdk, but again I'm new to all this, so I could be wrong.

https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/#ECS.DeleteService

"Deletes a specified service within a cluster. You can delete a service if you have no running tasks in it and the desired task count is zero. If the service is actively maintaining tasks, you cannot delete it, and you must update the service to a desired task count of zero. For more information, see UpdateService."

Reply:
If I remember correctly, we may not need to do that anymore. I think that was implemented as a workaround for ecs's unreliable api for deletes at the time. I believe that is fixed now.

Reply:
obvs it would be better to avoid adding needless code, but without a system test or full smoke testing, this will be a hard thing to confirm as fixed